common/src/main/java/com/microsoft/identity/common/internal/providers/oauth2/CurrentTaskAuthorizationActivity.java

```diff
@@ -0,0 +1,163 @@
+                mFragment.completeAuthorizationInBrowserFlow(getIntent().getStringExtra("RESPONSE_URI"));
+                finish();
+                return;
+            }
+
+            return;
+        }
+
+        setFragment(mFragment);
+
+        if (savedInstanceState == null) {
+
+            mCloseCustomTabs = false;
+
+            // This activity will receive a broadcast if it can't be opened from the back stack
+            redirectReceiver = new BroadcastReceiver() {
+                @Override
+                public void onReceive(Context context, Intent intent) {
+                    // Remove the custom tab on top of this activity.
+                    Intent newIntent = new Intent(CurrentTaskAuthorizationActivity.this, CurrentTaskAuthorizationActivity.class);
+                    newIntent.setAction(REFRESH_TO_CLOSE);
+                    newIntent.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP | Intent.FLAG_ACTIVITY_SINGLE_TOP);
+                    startActivity(newIntent);
+                }
+            };
+            LocalBroadcastManager.getInstance(this).registerReceiver(redirectReceiver,
+                    new IntentFilter(REDIRECT_RETURNED_ACTION)
+            );
+        }
+    }
+
+    @Override
+    public void onBackPressed() {
+        if (!mFragment.onBackPressed()) {
+            super.onBackPressed();
+        }
+    }
+
+    /**
+     * This is invoked when an existing activity is re-used and provided with a new intent with additional information
+     * NOTE: It's important that you use setIntent to update the intent associated with the activity.  Otherwise subsequent calls to
+     * getIntent() in other event handlers will return the original intent or null
+     * @param intent
+     */
+    @Override
+    protected void onNewIntent(Intent intent) {
+        super.onNewIntent(intent);
+        if (REFRESH_TO_CLOSE.equals(intent.getAction())) {
+            Intent broadcast = new Intent(DESTROY_REDIRECT_RECEIVING_ACTIVITY_ACTION);
+            LocalBroadcastManager.getInstance(this).sendBroadcast(broadcast);
+            unregisterAndFinish();
+        }
+        //IMPORTANT: If you don't call this...
+        //then getIntent will return the original intent used to launch the activity or null
+        setIntent(intent);
+    }
+
+    @Override
+    protected void onResume() {
+        super.onResume();
+
+        //onNewIntent will change the value of getIntent from what was used to create the activity
+        //to the intent that was communicated to an existing activity
+        if (REDIRECT_RETURNED_ACTION.equals(getIntent().getAction())) {
+
+            Bundle arguments = new Bundle();
+            arguments.putBoolean("RESPONSE", true);
+            mFragment.setArguments(arguments);
+            mFragment.completeAuthorizationInBrowserFlow(getIntent().getStringExtra("RESPONSE_URI"));
+            setResult(RESULT_OK);
+            unregisterAndFinish();
+
+        }
+        if (mCloseCustomTabs) {
```

Comment:
where is this one set to true (prior to line 156)?

Reply:
It's a member field.  I"ll add a comment to make this easier to follow.